addons/product/views/product_pricelist_views.xml

```diff
@@ -93,77 +93,79 @@
+                            </div>
+                        </group>
 
-                          <field name="price_round" string="Rounding Method"/>
-                          <field name="price_min_margin" string="Min. Margin"/>
-                          <field name="price_max_margin" string="Max. Margin"/>
-                      </group>
+                        <group name="pricelist_rule_base" groups="product.group_sale_pricelist">
+                            <group>
+                                <field name="fixed_price" widget="monetary" attrs="{'invisible':[('compute_price', '!=', 'fixed')]}"/>
+                                <label for="percent_price" string="Discount" attrs="{'invisible':[('compute_price', '!=', 'percentage')]}"/>
+                                <div class="o_row" attrs="{'invisible':[('compute_price', '!=', 'percentage')]}">
+                                    <field name="percent_price" class="oe_inline" attrs="{'invisible':[('compute_price', '!=', 'percentage')]}"/>%%
+                                </div>
+                                <field name="base" attrs="{'invisible':[('compute_price', '!=', 'formula')]}"/>
+                                <field name="base_pricelist_id" attrs="{
+                                    'invisible': ['|', ('compute_price', '!=', 'formula'), ('base', '!=', 'pricelist')],
+                                    'required': [('compute_price', '=', 'formula'), ('base', '=', 'pricelist')],
+                                    'readonly': [('base', '!=', 'pricelist')]}"/>
+                                <label for="price_discount" string="Discount" attrs="{'invisible':[('compute_price', '!=', 'formula')]}"/>
+                                <div class="o_row" attrs="{'invisible':[('compute_price', '!=', 'formula')]}">
+                                    <field name="price_discount"/>
+                                    <span>%%</span>
+                                </div>
+                                <field name="price_surcharge" widget="monetary" string="Extra Fee" attrs="{'invisible':[('compute_price', '!=', 'formula')]}"/>
+                                <field name="price_round" string="Rounding Method" attrs="{'invisible':[('compute_price', '!=', 'formula')]}"/>
+                                <label string="Margins" for="price_min_margin" attrs="{'invisible':[('compute_price', '!=', 'formula')]}"/>
+                                <div class="o_row" attrs="{'invisible':[('compute_price', '!=', 'formula')]}">
+                                    <field name="price_min_margin" string="Min. Margin" class="oe_inline" widget="monetary" nolabel="1"/>
+                                    <i class="fa fa-long-arrow-right mx-2 oe_edit_only" aria-label="Arrow icon" title="Arrow"/>
+                                    <field name="price_max_margin" string="Max. Margin" class="oe_inline" widget="monetary" nolabel="1"/>
+                                </div>
```

Comment:
Keep a separate group to display all 'formula' related fields:
```xml
<group name="pricelist_formula" attrs="{'invisible':[('compute_price', '!=', 'formula')]}">
...
```
That way you won't have to specify the same `attrs` for every field inside the group.